src/Microsoft.Azure.WebJobs.Host/Loggers/FunctionInstanceLogger.cs

```diff
@@ -24,10 +26,27 @@ public Task<string> LogFunctionStartedAsync(FunctionStartedMessage message, Canc
         {
             string traceMessage = string.Format(CultureInfo.InvariantCulture, "Executing '{0}' (Reason='{1}', Id={2})", message.Function.ShortName, message.FormatReason(), message.FunctionInstanceId);
             Log(LogLevel.Information, message.Function, message.FunctionInstanceId, traceMessage);
-
+            if (message.TriggerDetails != null && message.TriggerDetails.Count != 0)
+            {
+                LogTemplatizedTriggerDetails(message);
+            }
             return Task.FromResult<string>(null);
         }
 
+        private void LogTemplatizedTriggerDetails(FunctionStartedMessage message)
+        {
+            var templateKeys = message.TriggerDetails.Select(entry => $"{entry.Key}: {{{entry.Key}}}");
+            string messageTemplate = "Trigger Details: " + string.Join(", ", templateKeys);
+            string[] templateValues = message.TriggerDetails.Select(entry => $"{entry.Value}").ToArray();
```

Comment:
No need for the select here. Use: 
`string[] templateValues = message.TriggerDetails.Values.ToArray();`